clang/test/dpct/helper_files_ref/include/lapack_utils.hpp

```diff
@@ -909,6 +909,123 @@ oneapi::mkl::compz job2compz(const oneapi::mkl::job &job) {
+    throw std::runtime_error(
+        "The oneAPI Math Kernel Library (oneMKL) Interfaces "
+        "Project does not support this API.");
+#else
+    if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
+      device_ws_size =
+          oneapi::mkl::lapack::syev_scratchpad_size<T>(q, jobz, uplo, n, lda);
+    } else {
+      device_ws_size =
+          oneapi::mkl::lapack::heev_scratchpad_size<T>(q, jobz, uplo, n, lda);
+    }
+#endif
+  }
+};
+
+template <typename T> struct syheev_impl {
+  void operator()(sycl::queue &q, oneapi::mkl::compz jobz,
+                  oneapi::mkl::uplo uplo, std::int64_t n, void *a,
+                  std::int64_t lda, void *w, void *device_ws,
+                  std::size_t device_ws_size, int *info) {
+#ifndef __INTEL_MKL__
+    throw std::runtime_error(
+        "The oneAPI Math Kernel Library (oneMKL) Interfaces "
+        "Project does not support this API.");
+#else
+    using value_t = typename value_type_trait<T>::value_type;
+    auto a_data = dpct::detail::get_memory(reinterpret_cast<T *>(a));
+    auto device_ws_data =
+        dpct::detail::get_memory(reinterpret_cast<T *>(device_ws));
+    auto w_data = dpct::detail::get_memory(reinterpret_cast<value_t *>(w));
+    if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
+      oneapi::mkl::lapack::syev(q, jobz, uplo, n, a_data, lda, w_data,
+                                device_ws_data, device_ws_size);
+    } else {
+      oneapi::mkl::lapack::heev(q, jobz, uplo, n, a_data, lda, w_data,
+                                device_ws_data, device_ws_size);
+    }
+    dpct::detail::dpct_memset(q, info, 0, sizeof(int));
+#endif
+  }
+};
+
+template <typename T> struct syheevd_scratchpad_size_impl {
+  void operator()(sycl::queue &q, oneapi::mkl::job jobz, oneapi::mkl::uplo uplo,
+                  std::int64_t n, library_data_t a_type, std::int64_t lda,
+                  std::size_t &device_ws_size) {
+    if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
+      device_ws_size =
+          oneapi::mkl::lapack::syevd_scratchpad_size<T>(q, jobz, uplo, n, lda);
+    } else {
+      device_ws_size =
+          oneapi::mkl::lapack::heevd_scratchpad_size<T>(q, jobz, uplo, n, lda);
+    }
+  }
+};
+
+template <typename T> struct syheevd_impl {
+  void operator()(sycl::queue &q, oneapi::mkl::job jobz, oneapi::mkl::uplo uplo,
+                  std::int64_t n, library_data_t a_type, void *a,
+                  std::int64_t lda, void *w, void *device_ws,
+                  std::size_t device_ws_size, int *info) {
+    using value_t = typename value_type_trait<T>::value_type;
+    auto a_data = dpct::detail::get_memory(reinterpret_cast<T *>(a));
+    auto device_ws_data =
+        dpct::detail::get_memory(reinterpret_cast<T *>(device_ws));
+    auto w_data = dpct::detail::get_memory(reinterpret_cast<value_t *>(w));
+    if constexpr (std::is_same_v<T, float> || std::is_same_v<T, double>) {
```

Comment:
JFYI, it's also possible to use is_floating_point to check this condition.

Reply:
OK, I will update it. Thanks for the review.